### PR TITLE
feat: created theme provider and remove next.app.mjs

### PR DESCRIPTION
**Pull request by ovflowd**
## Description

This PR introduces the `themeProvider` for the `MUI` Theme Provider, `next-themes` Theme Provider, and `framer-motion` Motion Config.

It also applies the Font Family for the MUI Theme and `global jsx` style.

This PR then removes the shared `next.app.mjs` and has a dedicated app tree for the legacy Website `pages/_app.tsx` and the new website layouts that are on Storybook `.storybook/preview.tsx`

In the future we can recreate the shared `next.app.mjs` once that both `_app.tsx` and `preview.tsx` share the same base app tree.

## Related Issues

Relates to https://github.com/nodejs/nodejs.org/issues/5460

### CODE REVIEW

File: providers/themeProvider.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { ThemeProvider as NextThemeProvider } from 'next-themes';
+import { ThemeProvider as MUIProvider } from '@mui/material/styles';
+import { createTheme } from '@mui/material/styles';
+import { MotionConfig } from 'framer-motion';
+import { useMemo, type FC, type PropsWithChildren } from 'react';
```
Comment (AugustinMauroy) [inline providers/themeProvider.tsx]: wouldn't it be easier to read something like this?
```suggestion
import { useMemo } from 'react';
import type { FC, PropsWithChildren  } from 'react';
```

Comment (ovflowd) [reply providers/themeProvider.tsx]: I have no idea why the freak my commit made this.

Comment (ovflowd) [reply providers/themeProvider.tsx]: But this sounds like something new, didn't know you could import types like this nowadays hmmm...

**Review by AugustinMauroy — APPROVED** [review]
LGTM !

File: pages/_app.tsx [hunk 2]
```diff
@@ -1,17 +1,35 @@
...
 import type { AppProps } from 'next/app';
 
 import '../styles/old/index.scss';
 
-setAppFonts([sourceSans.style.fontFamily]);
-
 const App = ({ Component, pageProps }: AppProps) => (
-  <BaseApp>
-    <Component {...pageProps} />
+  <>
```
Comment (araujogui) [inline pages/_app.tsx:12]: Shouldn't `ThemeProvider` be here too?

File: pages/_app.tsx [hunk 3]
```diff
@@ -1,17 +1,35 @@
...
+            <Component {...pageProps} />
+          </BlogDataProvider>
+        </NodeReleasesProvider>
+      </SiteProvider>
+    </LocaleProvider>
+
     <Analytics />
-  </BaseApp>
+
+    <style jsx global>
```
Comment (araujogui) [inline pages/_app.tsx:25]: Style duplicate with `ThemeProvider`

**Review by araujogui — APPROVED** [review]

**Review by shanpriyan — APPROVED** [review]
LGTM!

**Pull request merged**